### ISSUE TITLE
Lambda count "drain" workers as "ready for scheduling"

### ISSUE THREAD

**Issue by MaxymVlasov** [spacelift-io/terraform-aws-spacelift-workerpool-on-ec2#84]
### How it works now

1. Let's imagine a situation when `Workers: 0, Busy: 0, Queue: 0`
2. Then, some run triggered and we get  `Workers: 0, Busy: 0, Queue: 1`
3. Lambda sees that `Queue > Workers`, create a new Worker, and run scheduled to it. We get  `Workers: 1, Busy: 1, Queue: 1`
4. After the run was completed, we had `Workers: 1, Busy: 0, Queue: 0` for some time.
5. Lambda sees that `Workers > Busy && Workers > Queue` so it starts termination of Worker drains it. 

So far so good. 

6. After the Worker drain, there is another run pop-up, so we have `Workers: 1, Busy: 0, Queue: 1`

BUT, because Worker is drained and  `Workers is not > Queue`, run in Queue end in long awaiting, till old one Worker will be terminated, and Lambda detect that `Queue > Workers` and create a new worked (step 3):

`Workers: 1, Busy: 0, Queue: 1` -> `Workers: 0, Busy: 0, Queue: 1` -> `Workers: 1, Busy: 1, Queue: 1` 

### Suggestion

In Lambda, Check the status of workers during the `Queue > Workers` evaluation.

Currently it is `Queue > all Workers`, when should be `Queue > not drain Workers`

**Comment by zhimsel** [comment]
This should probably be moved to https://github.com/spacelift-io/ec2-workerpool-autoscaler/issues, since this is an issue with the autoscaler itself, not this module.

**Comment by lorengordon** [comment]
> This should probably be moved to https://github.com/spacelift-io/ec2-workerpool-autoscaler/issues, since this is an issue with the autoscaler itself, not this module.

@zhimsel think GitHub has a transfer issue option, to move it to another repo.

**Comment by zhimsel** [comment]
> GitHub has a transfer issue option, to move it to another repo.

Yeah, pretty sure they do. The OP (@MaxymVlasov) needs to do that, though. Or maybe it needs to be a repo admin/maintainer on both sides.

**Comment by lorengordon** [comment]
@Apollorion ?

**Comment by Apollorion** [comment]
Github wont let me move it to that repo and only that repo. I could move it anywhere else. Ill notify one of our admins internally, maybe they can move it.
